**E-Commerce Data Analysis Assignment**

**1. Data Uploading**

In [ ]:
from google.colab import files
uploaded = files.upload()

Saving sales_dataset.csv to sales_dataset.csv


In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [ ]:
df = pd.read_csv('sales_dataset.csv')

/tmp/ipykernel_13523/2144452552.py:1: DtypeWarning: Columns (22) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv('sales_dataset.csv')


In [ ]:
df.head()
df.shape
df.columns
df.info()
df.describe()
df.isnull().sum()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 128949 entries, 0 to 128948
Data columns (total 23 columns):
 #   Column              Non-Null Count   Dtype  
---  ------              --------------   -----  
 0   Order ID            128949 non-null  object 
 1   Date                128949 non-null  object 
 2   Status              128949 non-null  object 
 3   Fulfilment          128949 non-null  object 
 4   Sales Channel       128949 non-null  object 
 5   ship-service-level  128949 non-null  object 
 6   Style               128949 non-null  object 
 7   SKU                 128949 non-null  object 
 8   Category            128949 non-null  object 
 9   Size                128949 non-null  object 
 10  ASIN                128949 non-null  object 
 11  Courier Status      122078 non-null  object 
 12  Qty                 128949 non-null  int64  
 13  currency            121155 non-null  object 
 14  Amount              121155 non-null  float64
 15  ship-city           128916 non-nul

,0
Order ID,0
Date,0
Status,0
Fulfilment,0
Sales Channel,0
ship-service-level,0
Style,0
SKU,0
Category,0
Size,0


**2. Data Cleaning**

In [ ]:
# Check missing values
df.isnull().sum()

,0
Order ID,0
Date,0
Status,0
Fulfilment,0
Sales Channel,0
ship-service-level,0
Style,0
SKU,0
Category,0
Size,0


In [ ]:
df['Amount'] = df['Amount'].fillna(df['Amount'].median())

In [ ]:
df.drop(columns=['Unnamed: 22'], inplace=True)

In [ ]:
df['Date'] = pd.to_datetime(df['Date'])

/tmp/ipykernel_13523/2394721818.py:1: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  df['Date'] = pd.to_datetime(df['Date'])


In [ ]:
df.duplicated().sum()

np.int64(6)

In [ ]:
df.drop_duplicates(inplace=True)

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 128943 entries, 0 to 128948
Data columns (total 22 columns):
 #   Column              Non-Null Count   Dtype         
---  ------              --------------   -----         
 0   Order ID            128943 non-null  object        
 1   Date                128943 non-null  datetime64[ns]
 2   Status              128943 non-null  object        
 3   Fulfilment          128943 non-null  object        
 4   Sales Channel       128943 non-null  object        
 5   ship-service-level  128943 non-null  object        
 6   Style               128943 non-null  object        
 7   SKU                 128943 non-null  object        
 8   Category            128943 non-null  object        
 9   Size                128943 non-null  object        
 10  ASIN                128943 non-null  object        
 11  Courier Status      122072 non-null  object        
 12  Qty                 128943 non-null  int64         
 13  currency            121152 non-nul

**3. Data Analysis**

In [ ]:
df['Date'] = pd.to_datetime(df['Date'])
df.groupby(df['Date'].dt.to_period('M'))['Amount'].sum() #Revenue growth over time

,Amount
Date,
2022-03,107128.85
2022-04,30646525.96
2022-05,27735050.75
2022-06,24797387.38


Insights:Revenue shows a sharp increase from March to April 2022, followed by consistently high sales in May and June, indicating strong business scaling during this period.

In [ ]:
df.groupby('Category')['Amount'].sum().sort_values(ascending=False) #top selling

,Amount
Category,
Set,41154789.67
kurta,23204584.70
Western Dress,11694003.69
Top,5623727.30
Ethnic Dress,830292.66
Blouse,485633.18
Bottom,162767.98
Saree,129378.76
Dupatta,915.00


Insights:“Set” and “Kurta” dominate sales, contributing the majority of revenue, showing strong demand for ethnic wear products.

In [ ]:
df['Amount'].sum() / df['Order ID'].nunique() #average order value

np.float64(692.020846683063)

Insights: The average order value of ₹692 indicates a moderately low basket size, suggesting high volume–low ticket sales strategy.

In [ ]:
df.groupby('ship-city')['Amount'].sum().sort_values(ascending=False).head(10) #Top locations

,Amount
ship-city,
BENGALURU,7167740.99
HYDERABAD,5227277.82
MUMBAI,3859728.80
NEW DELHI,3821817.78
CHENNAI,3273076.74
PUNE,2457729.18
KOLKATA,1497258.87
GURUGRAM,1272493.74
THANE,1061225.29


Insights: Bengaluru, Hyderabad, and Mumbai are the top revenue-generating cities, highlighting strong urban market concentration in metro cities.

In [ ]:
df[df['Status']=='Cancelled'].shape[0] / len(df) * 100 #%cancellation

14.2217879217949

Insights: A cancellation rate of 14.22% indicates moderate order leakage, suggesting scope for improving order confirmation and fulfillment processes.

In [ ]:
df.groupby('B2B')['Amount'].sum() / df['Amount'].sum() * 100 #b2b vs b2c

,Amount
B2B,
False,99.269793
True,0.730207


Insights:The business is highly B2C-driven (99.27%), showing strong dependence on individual customers rather than bulk business buyers.

In [ ]:
df['ship-service-level'].value_counts() #shipping performance

,count
ship-service-level,
Expedited,88590
Standard,40353


Insights: Expedited shipping is significantly higher than standard shipping, indicating customer preference for faster delivery services.

In [ ]:
df.groupby('Fulfilment')['Amount'].sum() / df['Amount'].sum() * 100 #amazon vs merchant

,Amount
Fulfilment,
Amazon,69.603569
Merchant,30.396431


Insights: Amazon handles nearly 70% of orders, showing strong reliance on in-house fulfillment for operational efficiency.

In [ ]:
df.groupby('Category')['Amount'].sum().sort_values(ascending=False) #high demand categories

,Amount
Category,
Set,41154789.67
kurta,23204584.70
Western Dress,11694003.69
Top,5623727.30
Ethnic Dress,830292.66
Blouse,485633.18
Bottom,162767.98
Saree,129378.76
Dupatta,915.00


Insights: “Set” and “Kurta” remain the most demanded categories, confirming consistent customer preference for ethnic wear products.

In [ ]:
df['Qty'].mean() #quantity analysis

np.float64(0.9046477901088078)

Insights: The average order quantity is low, indicating single-item purchase behavior dominates customer transactions.

In [ ]:
df['Status'].value_counts() #Return analysis

,count
Status,
Shipped,77764
Shipped - Delivered to Buyer,28771
Cancelled,18338
Shipped - Returned to Seller,1953
Shipped - Picked Up,973
Pending,658
Pending - Waiting for Pick Up,281
Shipped - Returning to Seller,145
Shipped - Out for Delivery,35


Insights: Most orders are successfully shipped and delivered, but cancellations and returns indicate operational inefficiencies and scope for improving customer satisfaction.


# **FINDINGS**
1. Revenue shows strong growth with peak performance in mid-2022  
2. Ethnic wear (Set & Kurta) dominates product demand  
3. Business is highly dependent on B2C customers  
4. Metro cities contribute majority of sales revenue  
5. Average order value is relatively low, indicating high-volume sales model  
6. Cancellation rate (14.22%) highlights operational leakage  
7. Amazon fulfillment dominates logistics operations  
8. Expedited shipping is preferred over standard delivery  
9. Returns and cancellations highlight need for better order management  



**RECOMMENDATIONS**
1. Focus marketing efforts on high-performing categories like Set and Kurta to maximize revenue and strengthen core product demand.  

2. Improve customer experience and order accuracy to reduce the 14.22% cancellation rate and minimize operational leakage.  

3. Increase Average Order Value by introducing product bundling, combo offers, and cross-selling strategies.  

4. Strengthen fulfillment partnerships and expand Amazon Fulfillment usage to ensure faster and more reliable deliveries.  

5. Optimize logistics for Standard Shipping to balance cost efficiency and delivery speed expectations.  

6. Target marketing campaigns in metro cities like Bengaluru, Hyderabad, and Mumbai to leverage high demand regions.  

7. Implement better inventory forecasting to avoid stock imbalances in high-demand categories.  

8. Analyze return and cancellation patterns to identify product or service issues and improve customer satisfaction.  

9. Encourage repeat purchases through loyalty programs and personalized marketing strategies.  

**CONCLUSION**
The e-commerce analysis highlights strong business performance driven primarily by B2C customers and high demand for ethnic wear products such as Set and Kurta. The company shows significant revenue concentration in metro cities and relies heavily on Amazon fulfillment and expedited shipping for efficient delivery operations.  

While the business demonstrates strong growth and operational efficiency, challenges such as moderate cancellation rates, low average order value, and return issues indicate areas for improvement.  

By optimizing inventory management, strengthening logistics systems, and enhancing customer experience strategies, the company can improve profitability, reduce operational inefficiencies, and achieve sustainable long-term growth in the competitive e-commerce market.  